lib/topological_inventory/ansible_tower/receptor/async_receiver.rb

```diff
@@ -0,0 +1,59 @@
+require "concurrent"
+require "topological_inventory/ansible_tower/logging"
+
+module TopologicalInventory::AnsibleTower
+  module Receptor
+    class AsyncReceiver
+      include Logging
+
+      attr_accessor :collector, :transformation
+      attr_reader :connection, :entity_type, :refresh_state_uuid, :refresh_state_started_at, :sweep_scope, :total_parts
+
+      def initialize(collector, connection, entity_type, refresh_state_uuid, refresh_state_started_at)
+        self.collector = collector
+        self.connection = connection
+        self.entity_type = entity_type
+        self.refresh_state_uuid = refresh_state_uuid
+        self.refresh_state_started_at = Time.now.utc
+        self.sweep_scope = Set.new
+        self.total_parts = Concurrent::AtomicFixnum.new
+        self.transformation = nil
+      end
+
+      def on_success(msg_id, entity)
+        # TODO: without tower hostname, it's not possible to construct job URL
+        parser = TopologicalInventory::AnsibleTower::Parser.new(tower_url: 'https://tower.example.com')
```

Comment:
Oh this is a fun problem. We probably won't even have the hostname on receptor-enabled towers, I wonder how we'll build the URL here.

Reply:
This issue was brought to my attention and something we'll need to discuss further, but should not be a blocker for this PR.  We discussed just returning the Tower job info (but not the entire URL) as well as Catalog ultimately using data returned from the Playbook set_stats to provide information to be displayed in the order.

The full URL was always more of a temporary work-around until we can get features like that into Catalog.